### PR TITLE
Use ??= more

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Bridge/PhpUnit/Legacy/ConstraintTraitForV7.php [hunk 1]
```diff
@@ -45,9 +45,7 @@ protected function additionalFailureDescription($other): string
 
     protected function exporter(): Exporter
     {
-        if (null === $this->exporter) {
-            $this->exporter = new Exporter();
-        }
+        $this->exporter ??= new Exporter();
```
Comment (derrabus) [inline src/Symfony/Bridge/PhpUnit/Legacy/ConstraintTraitForV7.php]: Shall we make this method a one-liner?

```php
return $this->exporter ??= new Exporter();
```

There are more methods like this one changed in this PR.

File: src/Symfony/Component/Config/Definition/Builder/ExprBuilder.php [hunk 2]
```diff
@@ -63,9 +63,7 @@ public function always(\Closure $then = null): static
      */
     public function ifTrue(\Closure $closure = null): static
     {
-        if (null === $closure) {
-            $closure = static function ($v) { return true === $v; };
-        }
+        $closure ??= static function ($v) { return true === $v; };
```
Comment (derrabus) [inline src/Symfony/Component/Config/Definition/Builder/ExprBuilder.php]: Maybe we should browse the codebase for potential arrow functions as well. 🤔 

```suggestion
        $closure ??= static fn ($v) => true === $v;
```

But this should probably be a separate PR.

File: src/Symfony/Component/DependencyInjection/Compiler/AbstractRecursivePass.php [hunk 3]
```diff
@@ -150,9 +150,7 @@ protected function getConstructor(Definition $definition, bool $required): ?\Ref
                 }
             } elseif ($class instanceof Definition) {
                 $class = $class->getClass();
-            } elseif (null === $class) {
-                $class = $definition->getClass();
-            }
+            } else$class ??= $definition->getClass();
```
Comment (derrabus) [inline src/Symfony/Component/DependencyInjection/Compiler/AbstractRecursivePass.php]: This looks broken.

**Review by derrabus — CHANGES_REQUESTED** [review]

**Comment by nicolas-grekas** [comment]
PR updated

**Review by derrabus — APPROVED** [review]

**Review by fancyweb — APPROVED** [review]

**Review by yceruto — APPROVED** [review]

**Pull request merged**